src/server/job_func.c

```diff
@@ -519,8 +519,10 @@ job_free(job *pj)
 			if ((pj->ji_wattr[JOB_ATR_euser].at_val.at_str) &&
 				(pwdp = getpwnam(pj->ji_wattr[JOB_ATR_euser].at_val.at_str))) {
 				if (pwdp->pw_userlogin != INVALID_HANDLE_VALUE) {
-					if (impersonate_user(pwdp->pw_userlogin) == 0)
+					if (impersonate_user(pwdp->pw_userlogin) == 0) {
+						log_joberr(-1, __func__, "impersonate_user() failed", pj->ji_qs.ji_jobid);
```

Comment:
same - dont u want to log the username for whom the impersonation failed?

Reply:
fixed in https://github.com/openpbs/openpbs/pull/1794/commits/90ed7603b8b003d419447d749bc57f0e30543573